src/FlatSharp.UnityPolyfills/UnityExtensions.cs

```diff
@@ -0,0 +1,96 @@
+﻿/*
+ * Copyright 2022 Unity Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace FlatSharp.Internal
+{
+    public static class UnityExtensions
+    {
+        public static NativeArray<TElement> ReadNativeArrayBlock<TElement, TBuffer>(this TBuffer buffer, int uoffset)
```

Comment:
> * Will generated code need to depend on this Polyfill assembly like it does FlatSharp.Runtime?

It will depend on `UnityExtensions.cs` only. The `NativeArray.cs` file will be fulfilled by assemblies present in the Unity environment.

> But in that case, we won't care which NativeArray is referenced, only that the type exists.
This should be the case today. Tests run with the `NativeArray` stubbed here. Unity uses it's own implementation.

> Would it perhaps be better to emit this from the Type Model codegen instead of a method here?
I have debated this, but I still need to resolve a `NativeArray` type. Since I already will require an external assembly reference for the `NativeArray` type, the hand written C# has not been too burdensome. Also, it's easier to reason about and iterate on as a static file ATM.

Reply:
Since UnityExtensions is in a different DLL, then we'll definitely need to push that as a separate package, correct?